geminimd/render.py

```
@@ -1,15 +1,71 @@
 
 
+def _human_size(num: Optional[int]) -> Optional[str]:
+    if num is None or num <= 0:
+        return None
+    units = ["B", "KB", "MB", "GB", "TB"]
+    idx = min(int(math.log(num, 1024)) if num > 0 else 0, len(units) - 1)
+    value = num / (1024 ** idx)
+    return f"{value:.2f} {units[idx]}"
+
+
+def _encode_metadata_value(value: Any) -> str:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)